### PR TITLE
Add active filter in stock management

### PR DESCRIPTION
**Pull request by aleeks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | BO: add active filter in stock management
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/FF-17
| How to test?  | play with filters in stock management

### CODE REVIEW

**Comment by AlexEven** [comment]
Wording:
I'm not comfortable with "Filter by enable status".

We should have something like

Filter by enable status => Filter by status
Disable => Disabled
Enable => Enabled
Enable Status => Status (like on Product page) or Enabled (like on 1.6 page)

Ping @xBorderie for a second opinion?

Thanks!

**Comment by xBorderie** [comment]
100% agreed with @AlexEven 's comment.

**Comment by nihco2** [comment]
@AlexEven @xBorderie Thanks, we change that

File: admin-dev/themes/new-theme/js/app/widgets/ps-radio.vue [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *-->
+
+<template>
+    <div class="radio">
```
Comment (aleeks) [inline admin-dev/themes/new-theme/js/app/widgets/ps-radio.vue]: Maybe this file hasn't the good indentation!

File: admin-dev/themes/new-theme/js/app/pages/stock/components/header/filters.vue [hunk 2]
```diff
@@ -76,6 +76,29 @@
             />
           </div>
         </div>
+        <div class="col-md-4">
+          <div class="p-y-2 p-x-2">
+            <h2>{{trans('filter_status')}}</h2>
+            <PSRadio
+                :label="trans('filter_status_enable')"
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/js/app/pages/stock/components/header/filters.vue]: Check the indent

File: admin-dev/themes/new-theme/js/app/pages/stock/components/overview/products-table.vue [hunk 3]
```diff
@@ -39,6 +39,9 @@
           <PSSort order="supplier" @sort="toggleSort" />
         </th>
         <th class="text-xs-center">
+           {{trans('title_status')}}
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/js/app/pages/stock/components/overview/products-table.vue]: Check the indent

File: admin-dev/themes/new-theme/js/app/widgets/ps-radio.vue (hunk 1 above)
Comment (maximebiloe) [reply admin-dev/themes/new-theme/js/app/widgets/ps-radio.vue]: Yes, all the file is missindented

File: src/PrestaShopBundle/Api/QueryParamsCollection.php [hunk 4]
```diff
@@ -315,6 +315,10 @@ protected function appendSqlFilter($value, $column, array $filters)
             return $this->appendSqlDateAddFilter($filters, $value);
         }
 
+        if ($column === 'active') {
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Api/QueryParamsCollection.php]: Yoda condition 😉

File: src/PrestaShopBundle/Translation/Api/StockApi.php [hunk 5]
```diff
@@ -46,6 +46,10 @@ public function getTranslations()
             'filter_movements_period' => $this->translator->trans('Filter by period', array(), 'Admin.Catalog.Feature'),
             'filter_search_category' => $this->translator->trans('Search a category', array(), 'Admin.Catalog.Feature'),
             'filter_search_suppliers' => $this->translator->trans('Search a supplier', array(), 'Admin.Catalog.Feature'),
+            'filter_status' => $this->translator->trans('Filter by enable status', array(), 'Admin.Catalog.Feature'),
+            'filter_status_all' => $this->translator->trans('All', array(), 'Admin.Catalog.Feature'),
+            'filter_status_disable' => $this->translator->trans('Disable', array(), 'Admin.Catalog.Feature'),
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Translation/Api/StockApi.php]: Admin.Actions

File: src/PrestaShopBundle/Translation/Api/StockApi.php [hunk 6]
```diff
@@ -46,6 +46,10 @@ public function getTranslations()
             'filter_movements_period' => $this->translator->trans('Filter by period', array(), 'Admin.Catalog.Feature'),
             'filter_search_category' => $this->translator->trans('Search a category', array(), 'Admin.Catalog.Feature'),
             'filter_search_suppliers' => $this->translator->trans('Search a supplier', array(), 'Admin.Catalog.Feature'),
+            'filter_status' => $this->translator->trans('Filter by enable status', array(), 'Admin.Catalog.Feature'),
+            'filter_status_all' => $this->translator->trans('All', array(), 'Admin.Catalog.Feature'),
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Translation/Api/StockApi.php]: Admin.Actions

File: src/PrestaShopBundle/Translation/Api/StockApi.php [hunk 7]
```diff
@@ -46,6 +46,10 @@ public function getTranslations()
...
+            'filter_status_disable' => $this->translator->trans('Disable', array(), 'Admin.Catalog.Feature'),
+            'filter_status_enable' => $this->translator->trans('Enable', array(), 'Admin.Catalog.Feature'),
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Translation/Api/StockApi.php]: Admin.Actions

**Review by maximebiloe — CHANGES_REQUESTED** [review]
For the indent, please make sure to have the editorconfig plugin enable and the ESLint

File: admin-dev/themes/new-theme/js/app/pages/stock/components/header/filters.vue [hunk 8]
```diff
@@ -188,6 +219,7 @@
     border-radius: 2px;
     box-shadow: 1px 2px 3px 0 rgba(108, 134, 142, 0.3);
     border: solid 1px #b9cdd2;
+    width: 150%;
```
Comment (aleeks) [inline admin-dev/themes/new-theme/js/app/pages/stock/components/header/filters.vue:225]: sure about this ?

**Comment by aleeks** [comment]
done @maximebiloe

**Review by maximebiloe — DISMISSED** [review]

**Comment by emmanuelf75009** [comment]
Ok on my side

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @aleeks and @nihco2!

**Pull request merged**